### PR TITLE
Moving remove cli till after gen metadata and only if failure

### PR DESCRIPTION
**Pull request by paigerube14**
### Description
Moving remove_cli to after gen metadata so the python virtualenv is still avaiable with pyyaml package. 

Also adding if statement if project_name does not exist in the build helper (was seeing the error under fixes)

### Fixes
https://github.com/cloud-bulldozer/e2e-benchmarking/issues/371

```
04-07 11:40:36.224  No resources found in --no-headers namespace.
04-07 11:40:36.224  build_helper.sh: line 10: ${#project_name}-1: substring expression < 0
```

### CODE REVIEW

File: utils/benchmark-operator.sh [hunk 1]
```diff
@@ -65,9 +65,12 @@ run_benchmark() {
       kubectl logs -n benchmark-operator --prefix --tail=30 ${pod}
       kubectl logs -n benchmark-operator --prefix --tail=-1 ${pod} >> ${pod_log}
     done
-    remove_cli
   fi
   local benchmark_name=$(cat ${1} | python -c 'import yaml; import sys; print(yaml.safe_load(sys.stdin.read())["metadata"]["name"])')
   gen_metadata ${benchmark_name} ${start_date} $(date +%s%3N)
+
+  if [[ -${rc} -eq 1 ]]; then
```
Comment (rsevilla87) [inline utils/benchmark-operator.sh:72]: Why did you add this conditional?

Comment (paigerube14) [reply utils/benchmark-operator.sh:72]: because the remove_cli was originally only in the if statement starting on line 57, so I was trying to keep the same functionality by putting an if statement

**Comment by paigerube14** [comment]
BTW I think the second issue I listed under fixes is actually fixed by not running the incorrect workload ([PR 373](https://github.com/cloud-bulldozer/e2e-benchmarking/pull/373))

But still think the extra check for project name is helpful in the build_helper.py file

File: utils/benchmark-operator.sh (hunk 1 above)
Comment (paigerube14) [reply utils/benchmark-operator.sh]: @rsevilla87 do you think we could move the _remove_cli_ to the end of the run.sh scripts in case other places need items installed by the benchmark operator requirements? There might also be other cases similar to the one I hit if we remove the cli but need a package that was installed in the virtualenv

Do we need to be calling remove_benchmark_operator instead?

Comment (paigerube14) [reply utils/benchmark-operator.sh]: Specifically for the concurrent-builds it runs _run_benchmark_ multiple times so I would have to add in a call to re-activate the cli

**Review by rsevilla87 — APPROVED** [review]
lgtm!
